nixos/modules/services/security/bitwarden_rs/default.nix

```diff
@@ -0,0 +1,177 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.bitwarden_rs;
+  description = "Bitwarden Server (Rust Edition)";
+
+  nameToKey = name:
+    let
+      parts = builtins.split "([A-Z0-9]+)" name;
+      partsToKey = parts: removePrefix "_" (foldl' (key: x:
+        if isList x then key + "_" + head x
+        else if key != "" && elem (substring 0 1 x) lowerChars then # to handle e.g. [ "disable" [ "2FAR" ] "emember" ]
+          let last = stringLength key - 1;
+          in substring 0 last key + optionalString (substring (last - 1) 1 key != "_") "_" + substring last 1 key + toUpper x
+        else key + toUpper x) "" parts);
+    in partsToKey parts;
+
+  configFile = pkgs.writeText "bitwarden_rs.env" (concatMapStrings (s: s + "\n") (
+    (concatLists (mapAttrsToList (name: value:
+      if value != null then [ "${nameToKey name}=${if isBool value then boolToString value else toString value}" ] else []
+    ) (cfg.config // {
+      inherit (cfg) dataFolder;
+      webVaultEnabled = cfg.enableWebVault;
+      webSocketEnabled = cfg.enableWebsocket;
+    })))));
+
+in {
+  options.services.bitwarden_rs = with types; {
+    enable = mkEnableOption description;
+
+    user = mkOption {
+      type = str;
```

Comment:
This is a bit of a personal taste, but I like to avoid such options when not necessary by doing something like

```nix
let
  user = config.users.users.bitwarden_rs.name;
  group = config.users.groups.bitwarden_rs.name;
in {
  # ...
  config = mkIf cfg.enable {
    users.users.bitwarden_rs.group = group;
    users.bitwarden_rs.murmur = {};

    systemd.services.bitwarden_rs.serviceConfig = {
      User = user;
      Group = group;
    };
    # ...
  };
}
```

This doesn't introduce any additional options, but also doesn't hardcode the user, because it still lets you do this to override it:
```nix
{
  users.users.bitwarden_rs.name = "paul";
}
```

And as a bonus, it alleviates the need for eerie `optional (cfg.user == "bitwarden_rs")` user creation conditions.

Whatever suits you best here.